bgpd/bgp_route.c

```diff
@@ -5670,16 +5690,43 @@ static void bgp_aggregate_route(struct bgp *bgp, struct prefix *p,
 			} else
 				aspath = aspath_dup(pi->attr->aspath);
 
-			if (!pi->attr->community)
-				continue;
+			if (pi->attr->community) {
+				if (community) {
+					commerge = community_merge(
+						community, pi->attr->community);
+					community =
+						community_uniq_sort(commerge);
+					community_free(commerge);
```

Comment:
The signature for community_free is different from lcommunity_free & ecommunity_free. The code same to exactly same except the type passed to free call. I think we can fix and this unnecessary confusion can be avoided. 

Code changes looks good.